packages/api/schema/stryker-core.json

```diff
@@ -400,6 +400,14 @@
       "type": "string",
       "default": "command"
     },
+    "testRunnerNodeArgs": {
+      "description": "Configure arguments to be passed as exec arguments to the test runner child process. For example, running Stryker with `--timeoutMS 9999999 --concurrency 1 --testRunnerNodeArgs --inspect-brk` will allow you to debug the test runner child process. See `execArgv` of [`child_process.fork`](https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options)",
```

Comment:
It's probably a good idea to support and communicate with quotes around the test runner args:

```
--timeoutMS 9999999 --concurrency 1 --testRunnerNodeArgs "--inspect-brk"
```

This allows us to send multiple args like this:

```
--timeoutMS 9999999 --concurrency 1 --testRunnerNodeArgs "--inspect-brk --foo --bar --baz"
```

It also makes it clearer which args are for the main process and which for the test runner.

Reply:
The latter results in `process.argv` containing:

```
[
  '--timeoutMS',
  '9999999',
  '--concurrency',
  '1',
  '--testRunnerNodeArgs',
  '--inspect-brk --foo --bar --baz'
]
```

So `commander` should be able to handle it without issues.

Reply:
> The latter results in `process.argv` containing:
> 
> ```
> [
>   '--timeoutMS',
>   '9999999',
>   '--concurrency',
>   '1',
>   '--testRunnerNodeArgs',
>   '--inspect-brk --foo --bar --baz'
> ]
> ```
> 
> So `commander` should be able to handle it without issues.

Well. We currently support splitting on "," as you can see here: 

https://github.com/stryker-mutator/stryker/blob/b2e541badd08e34826bc0ac6338cfb40f2e8565d/packages/core/src/stryker-cli.ts#L23

I've added an example with 2 arguments in the readme: `--testRunnerNodeArgs --inspect-brk,--cpu-prof`. Would this be acceptable?

> Don't forget to gracefully exit the child processes as I did here: https://github.com/stryker-mutator/stryker/pull/2536/files#diff-9bd013c2c0d5a7d2bcd8594a6bfb8b4a89b54b3db0d4626b61637f00fd37900d
> 
> Without it, things that trigger on `process.exit` will not trigger. The result of this is that, for example, `--cpu-prof` does not output anything.

Hmm, interesting. I don't seem to get CPU profile files at all this way. Even with the `finally(() => process.exit());`. I'm using Linux, this kind of behavior will definitly be influenced by OS specific closing details. Needs further investigating. I'm hesitant to drag it into this PR, since it is not directly related.

Reply:
Splitting on `','` would also work, but splitting on `/\s+/` (= any whitespace of any length) would make it feel a lot more like a command to me. If I didn't read the docs I would try `--testRunnerNodeArgs "--foo --bar"` before trying `--testRunnerNodeArgs --foo,--bar`. Both options will work for me though :)

> this kind of behavior will definitly be influenced by OS specific closing details. Needs further investigating.

If you add this to the PR I'll be able to test it on Windows and MacOS. With you testing on Linux we can cover all bases. This might be out of scope for this PR though, we can also cover these bases separately. Also note that `--cpu-prof` is experimental at this time: https://nodejs.org/api/cli.html#cli_cpu_prof. There might be stuff that doesn't work on every OS because of it.

Reply:
The "problem" with splitting on whitespace characters is that there are exotic cases of command-line arguments that require spaces. Not sure about node args though.

I've chosen `,` to split, because we're already using that for other options, like `--mutate` or `--files` (spaces in file names are a bit more common). I could change that for `--testRunnerNodeArgs` to also except a space. That might make sense as you point out.

Reply:
Yeah, you're right. There are some exotic cases that can pop up. This is generally the moment where I wonder if I can fix all of this with NPM :)

How about using something like [`string-argv`](https://www.npmjs.com/package/string-argv) or [`spawn-args`](https://www.npmjs.com/package/spawn-args) to prevent the pain of edge cases?

Reply:
I've chosen to split on `" "` (space) for now. I've updated the docs and code. It's a better user experience than splitting on a comma.

Don't want to add an extra dependency if I don't have to. I think splitting on spaces for node-args is fine (don't see a direct reason to support escaped spaces)

Reply:
Thanks for helping this feature!

Reply:
No problem :)